larq_compute_engine/mlir/ir/lce_ops.cc

```diff
@@ -73,5 +77,12 @@ void LarqDialect::initialize() {
 #include "larq_compute_engine/mlir/ir/lce_ops.cc.inc"
       >();
 }
+
+Operation* LarqDialect::materializeConstant(OpBuilder& builder, Attribute value,
+                                            Type type, Location loc) {
+  if (arith::ConstantOp::isBuildableWith(value, type))
+    return builder.create<mlir::arith::ConstantOp>(loc, type, value);
+  return nullptr;
+}
```

Comment:
In the tensorflow example they first try to make it into a `TFL::ConstOp` before they try the `arith::ConstOp`. I guess we don't need that here because the larq dialect is used for TF ops and not TFL ops?

Reply:
In the tensorflow example they first try to make it into a `TFL::ConstOp` before they try the `arith::ConstOp`. I guess we don't need that here because the larq dialect is used for TF ops and not TFL ops?

Reply:
If I understand their code correctly it will create a `tfl.pseudo_const` if the [value is opaque](https://mlir.llvm.org/docs/Dialects/Builtin/#opaqueelementsattr) which isn't supported by `arith::ConstantOp`. I don't think this is a case that we will encounter. In general `TFL::ConstOp` will always be [folded to `arith::ConstantOp` if possible](https://github.com/tensorflow/tensorflow/blob/3f878cff5b698b82eea85db2b60d65a2e320850e/tensorflow/compiler/mlir/lite/ir/tfl_ops.cc#L2770-L2794) so I think this should be fine to keep it like this for now.